kubernetes-repositories.md

```diff
@@ -0,0 +1,106 @@
+## Kubernetes Repositories
+
+
+This document attempts to outline a structure for creating and associating github repositories with the Kubernetes project.
+
+The document presents a tiered system of repositories with increasingly strict requirements in an attempt to provide the right level of oversight and flexibility for a variety of different projects.
+
+
+### Associated Repositories
+
+Associated repositories conform to the Kubernetes community standards for a repository, but otherwise have no restrictions. Associated repositories exist solely for the purpose of making it easier for the Kubernetes community to work together. There is no implication of support or endorsement of any kind by the Kubernetes project, the goals are purely logistical.
+
+#### Goals
+
+To facilitate contributions and collaboration from the broader Kubernetes community.  Contributions to random projects with random CLAs (or DCOs) can be logistically difficult, so associated repositories should be easier.
+
+
+#### Rules
+
+   * Must adopt the Kubernetes Code of Conduct statement in their repo.
+   * All code projects use the Apache License version 2.0. Documentation repositories must use the Creative Commons License version 4.0.
+   * Must adopt the CNCF CLA bot automation for pull requests.
+
+
+### SIG repositories
+
+SIG repositories serve as temporary homes for SIG-sponsored experimental projects or prototypes of new core functionality, or as permanent homes for SIG-specific tools.
+
+#### Goals
+
+To provide a place for SIGs to collaborate on projects endorsed by and actively worked on by members of the SIG. SIGs should be able to approve and create new repositories for SIG-sponsored projects without requiring higher level approval from a central body (e.g. steering committee or sig-architecture)
+
+#### Rules
+
+   * For now all repos will live in github.com/kubernetes-sigs/\<sig-name\>_\<project-name\>, this may change in the future as we see how things work.
```

Comment:
I agree with the email comment that this naming convention is a ux disaster. I can think of two improvements:
1. At least drop the "sig" from the sig name, e.g. `kubernetes-sigs/api-machinery_foobar` rather than `kubernetes-sigs/sig-api-machinery_foobar`
2. Give each sig a 2, 3, or 4 letter code. Then we can also switch to a dash rather than mix dashes and underscores. E.g., `kubernetes-sigs/apim-foobar`. Tracking short names for all sigs would be useful in other contexts, too, I think.

Reply:
Agree on UX disaster.

Two other alternatives: 
- put sig name  entirely from the repo name and move it to the repo _title_ (which can be known from GH API).  E.g. `kubernetes-sigs/foobar` with title `foobar: Foos your bars, maintained by #sig-api-machinery` 
- put entire sig name in a well known file such as `.prow/signame`. e.g. repo is called `kubernetes-sigs/foobar` and contains `.prow/signame` with contents `sig-api-machinery`.

Reply:
Another alternative is to use GitHub topics; github allows searching topics within an org, which is better than just use prefixes.